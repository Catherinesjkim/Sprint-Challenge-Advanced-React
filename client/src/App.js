// 1. Class component that fetches data from the server you now have running - the data can be fetched from `http://localhost:5000/api/players`
// 2. Display the player data you receive from the API

import React from 'react';
import './App.css';
import Navbar from './components/Navbar';
import Players from './components/Players';

class App extends React.Component {
    constructor(props) {
      super(props);
      this.state = {
        data: []
      };
    }

    componentDidMount() {
      fetch('http://localhost:5000/api/players')
      .then(response => response.json())
      .then(data => this.setState({ ...this.state, data: data }))
    }

    render () {
      return (
        <div className="App" data-testide="toggle">
          <Navbar />
          <div className="header">
            <h1>Players</h1>
          </div>
          <div>
            {this.state.data.map(info => {

              return (
                <Players 
                name={info.name}
                country={info.country}
                searches={info.searches}
                />
              )
            })}
          </div>
        </div>
    );
  }
}

export default App;
