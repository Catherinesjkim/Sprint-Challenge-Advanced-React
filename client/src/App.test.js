import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import * as rtl from '@testing-library/react';
import { toBeInTheDocument, toHaveClass } from '@testing-library/jest-dom';
expect.extend({toBeInTheDocument, toHaveClass})
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  rtl.render(<App />, div);
});

// First unit test: classname test
// test("It has a classname toggle - container", () => {
//   const{ getByTestId } = rtl.render(<App />);
//   expect(getByTestId("toggle")).toBeInTheDocument();
//   expect(getByTestId("toggle")).toHaveClass("App");
// });

// Second test: snapshot test to ensure our UI doesn't change without the team knowing about it. 
test("renders correctly when there are no players", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Third test: Integration test - tests interactiosn with the user
test("container players title", () => {
  // ARRANGE
  const expectedPlayersElements = 2;
  const { getAllByText } = rtl.render(<App />);

  // ACT
  const playersElements = getAllByText(/players/i);

  // ASSERT
  expect(playersElements.length).toBe(expectedPlayersElements);
});