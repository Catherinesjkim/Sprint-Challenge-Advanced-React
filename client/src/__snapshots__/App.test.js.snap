// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no players 1`] = `
<div>
  <h1 />
  <h2 />
  <h3 />
</div>
`;
